SofaKernel/modules/SofaEngine/BoxROI.inl

```diff
@@ -591,143 +591,142 @@ void BoxROI<DataTypes>::doUpdate()
     WriteOnlyAccessor< Data<vector<Hexa> > > hexahedraInROI = d_hexahedraInROI;
     WriteOnlyAccessor< Data<vector<Quad> > > quadInROI = d_quadInROI;
 
-
-    // Clear lists
-    indices.clear();
-    edgeIndices.clear();
-    triangleIndices.clear();
-    tetrahedronIndices.clear();
-    hexahedronIndices.clear();
-    quadIndices.clear();
+    if(d_doUpdate.getValue())
```

Comment:
I think that the if(d_doUpdate.getValue()) should be at the begining of the method like that:
```void doUpdate(){
    if(!d_doUpdate.getValue())
             return
```
Because there is no need have extra line if nothing is updated. 
@hugtalbot I don't see the point with invalid state.